kube-aws-updater

```diff
@@ -0,0 +1,160 @@
+    done
+  fi
+}
+
+label_for_cycling() {
+  role=${1}
+  nodes=$(kubectl --context=${KUBE_CONTEXT} get nodes -l role=${role} -o json | jq -r '.items[].metadata.name')
+  echo "${KUBE_CONTEXT} cluster has $(echo "${nodes}" | wc -l) ${role}s"
+  echo "Labelling ${role}s for retirement..."
+  while read -r node
+  do
+    kubectl --context=${KUBE_CONTEXT} label node ${node} retiring=${RETIRE_TIME} --overwrite=true
+    kubectl --context=${KUBE_CONTEXT} cordon ${node}
+  done <<< "${nodes}"
+}
+
+kill_node() {
+  node=${1}
+
+  echo "---------- ${node} ----------"
+  echo "draining ${node}..."
+  time kubectl --context=${KUBE_CONTEXT} drain ${node} --ignore-daemonsets --force --delete-local-data
+  instance_id=$(aws ${AWS_PROFILE_CLI} --output=json ec2 describe-instances --filters "Name=network-interface.private-dns-name,Values=${node}" \
+    | jq -r '.Reservations[].Instances[].InstanceId')
+  echo "terminating ${node} with instance-id ${instance_id}"
+  time aws ${AWS_PROFILE_CLI} ec2 terminate-instances --instance-ids=${instance_id}
+  echo "----------------------------------------------------------------------"
+}
+
+cycle_nodes() {
+  role=${1}
+
+  # - double ASG size
+  instance_address=$(kubectl --context=${KUBE_CONTEXT} get nodes -l role=${role},retiring=${RETIRE_TIME} -o json | jq -r '.items[0].metadata.name')
+  instance_id=$(aws ${AWS_PROFILE_CLI} --output=json ec2 describe-instances --filters "Name=network-interface.private-dns-name,Values=${instance_address}" \
+    | jq -r '.Reservations[].Instances[].InstanceId')
+  asg_name=$(aws ${AWS_PROFILE_CLI} --output=json autoscaling describe-auto-scaling-groups \
+    | jq -r '."AutoScalingGroups"[] | select(."Instances"[]."InstanceId"=="'${instance_id}'") | ."AutoScalingGroupName"')
+  asg_count=$(aws ${AWS_PROFILE_CLI} --output=json autoscaling describe-auto-scaling-groups \
+    | jq -r '."AutoScalingGroups"[] | select(."Instances"[]."InstanceId"=="'${instance_id}'") | ."DesiredCapacity"')
+  aws ${AWS_PROFILE_CLI} --output=json autoscaling update-auto-scaling-group --auto-scaling-group-name "${asg_name}" --desired-capacity $(( ${asg_count} * 2)) --max-size $(( ${asg_count} * 2))
+
+  # - wait for 2x group
+  count=${asg_count}
+  while [[ ${count} -lt $(( ${asg_count} * 2)) ]]
+  do
+    echo "${count}/$(( ${asg_count} * 2)) ready ${role}s"
+    sleep 8
+    count=$(kubectl --context=${KUBE_CONTEXT} get node -lrole=${role} | grep -v NotReady | tail -n +2 | wc -l)
+  done
+
+  # - stop ASG actions
+  aws ${AWS_PROFILE_CLI} --output=json autoscaling suspend-processes --auto-scaling-group-name "${asg_name}"
+
+  ### - move the registry
+  if [[ ${role} == "worker" ]]; then
+    handle_registry
+  fi
+
+  # - drain/terminate labelled masters (sequentially)
+  old_nodes=$(kubectl --context=${KUBE_CONTEXT} get nodes -l role=${role},retiring=${RETIRE_TIME} -o json | jq -r '.items[].metadata.name')
+  while read -r old_node
```

Comment:
Use consistent syle. Put ; do and ; then on the same line as the while, for or if.